test-data/unit/check-python310.test

```diff
@@ -567,6 +567,20 @@ match m:
         reveal_type(m)  # N: Revealed type is "builtins.tuple[Any, ...]"
 [builtins fixtures/primitives.pyi]
 
+[case testMatchInvalidClassPattern]
+m: object
+
+match m:
+    case xyz(y):  # E: Name "xyz" is not defined
+        reveal_type(m)  # N: Revealed type is "Any"
+        reveal_type(y)  # E: Cannot determine type of "y" \
```

Comment:
"Cannot determine type" is a bit unfortunate; I'd rather it's just Any. Probably fine this way though.

Reply:
Yeah, Any would be better, but I think that this would require a much bigger change.